packages/common/src/services/auth/hedgehog.ts

```diff
@@ -25,21 +29,83 @@ export type HedgehogInstance = Hedgehog & {
 }
 
 export const createHedgehog = ({
-  identityService,
+  identityServiceEndpoint,
   useLocalStorage = true,
   localStorage,
   createKey = getPlatformCreateKey()
 }: HedgehogConfig): HedgehogInstance => {
-  const getFn: IdentityService['getFn'] = async (obj) => {
-    return await identityService.getFn(obj)
+  const makeIdentityRequest = async <T = unknown>(
+    axiosRequestObj: AxiosRequestConfig
+  ) => {
+    axiosRequestObj.baseURL = identityServiceEndpoint
+
+    const requestId = uuid()
+    axiosRequestObj.headers = {
+      ...(axiosRequestObj.headers || {}),
+      'X-Request-ID': requestId
+    }
+
+    // Axios throws for non-200 responses
+    try {
+      const resp: AxiosResponse<T> = await axios(axiosRequestObj)
+      if (!resp.data) {
+        throw new Error(
+          `Identity response missing data field for url: ${axiosRequestObj.url}, req-id: ${requestId}`
+        )
+      }
+      return resp.data
+    } catch (e) {
+      const error = e as AxiosError
+      if (error.response?.data?.error) {
+        console.error(
+          `Server returned error for requestId ${requestId}: [${error.response.status.toString()}] ${
+            error.response.data.error
+          }`
+        )
+      }
+      throw error
+    }
   }
 
-  const setAuthFn: SetAuthFn = async (obj) => {
-    return await identityService.setAuthFn(obj)
+  const getFn: GetFn = async (params) => {
+    return await makeIdentityRequest({
+      url: '/authentication',
+      method: 'GET',
+      params
+    })
   }
 
-  const setUserFn: SetUserFn = async (obj) => {
-    return await identityService.setUserFn(obj)
+  const setAuthFn: SetAuthFn = async (params) => {
+    // get wallet from hedgehog and set as owner wallet
+    const ownerWallet = params.wallet
+    // delete wallet object so it's not passed to identity
+    // @ts-ignore
+    delete params.wallet
```

Comment:
is this safe? kinda feels scary. can we spread without instead?

Reply:
Oh duh, spread is a much better idea. I just moved this code without thinking about it too much. Will update.